### ISSUE TITLE
Publication of the Fedora ontology

### ISSUE THREAD

**Issue by acoburn** [fcrepo/fcrepo-specification#108]

**Comment by barmintor** [comment]
I can write up the stub page for this, but I need help publishing it to that server @awoods. This seems very important.

**Comment by barmintor** [comment]
@awoods for example, maybe we could set up another repo to be published to /definitions?

**Comment by acoburn** [comment]
@barmintor I have some repositories in which I write an ontology in TTL format and then use a python script to generate the JSON-LD, RDF/XML and HTML versions. A simple server config can then be used to handle conneg. See this for an example: https://github.com/trellis-ldp/trellis-ontology (the python code is in `./scripts`).

**Comment by awoods** [comment]
All of thhe current ontologies can currently be found here:
http://fedora.info/ (e.g. http://fedora.info/definitions/v4/repository#)

In order to avoid confusion, we will want to reconcile these with the proposed new location: http://fedora.info/definitions/repository#

..a simple approach will be to redirect `definitions/v4/repository` to `definitions/repository` and leave the existing ontologies where they are.

**Comment by acoburn** [comment]
@awoods can you clarify what you mean here? I really don't understand. Are you suggesting that the Fedora4 ontology simply redirects to the new location? If so what does that mean for all of the terms currently defined there but which are not part of this new ontology. Such a redirect would, if anything, cause more confusion. As I understand this new ontology, it is completely separate from the F4 ontology.

**Comment by acoburn** [comment]
To be clear, as long as the current terms in the Fedora4 ontology do _not_ go into this new ontology (e.g. fedora:PairTree, fedora:Tombstone, etc) I don't actually have an opinion about what is done with the F4 ontology. I do feel strongly however that any terms in the new ontology need strong justification for being there: e.g. the term is defined in the specification.

**Comment by escowles** [comment]
I don't think we can just redirect `definitions/v4/repository` to `definitions/repository` because the Modeshape impl. uses a number of terms that will not be included in `definitions/repository`.  Some of those terms can be replaced by `definition/repository` terms, and we should do that as we align the Modeshape impl. with the spec.

So I think this is still a valid issue.  We should develop a process for publishing whatever terms we put in https://github.com/fcrepo/fcrepo-specification/blob/master/repository.ttl

**Comment by awoods** [comment]
Publishing the 
1. [core Fedora repository ontology](https://github.com/fcrepo/fcrepo-specification/blob/master/repository.ttl) in parallel with the 
2. [community implementation ontology](http://fedora.info/definitions/v4/repository#) and
3. other implementation ontologies, makes sense.

Since the "[preferred prefix](https://github.com/fcrepo/fcrepo-specification/blob/master/repository.ttl#L16)" for the first two ontologies above is `fedora`, I would suggest we establish a new preferred prefix for the core Fedora repository ontology... maybe `fedoracore`.

@escowles : in terms of process, are you thinking of the mechanics of keeping the published ontology current or the process by which the content of the ontology is updated? If the latter, I would propose that we keep the ontology update process in line with the specification update process.

**Comment by acoburn** [comment]
@awoods for the reasons already outlined in #122 I am 👎  on using the prefix `fedoracore` for the Fedora ontology. I think the Fedora ontology should have a preferred prefix of `fedora`. Implementations should take appropriate measures to make sure there are no namespace conflicts. For instance, the community implementation may choose a different prefix, such as `fcrepo` or `fedora4` or `fedoramodeshape`.

As for the statement "the 'preferred prefix' for the first two ontologies above is `fedora`", can you explain that? That is certainly the case for the Fedora API ontology via `vann:preferredNamespacePrefix`, but there is no triple related to any such `vann:preferredNamespacePrefix` in the Fedora 4 ontology. The use of `fedora` is merely convention.

**Comment by ajs6f** [comment]
It would be inappropriate for the Modeshape implementation to arrogate `fedora` as the "preferred prefix" for a vocabulary specific to itself, by a particular predicate or by convention. It would be a small but meaningful blow against all the work that has been done to create a larger set of possibilities for the Fedora community. `fedoramodeshape` or something like it is not only more appropriate, but more accurate.

**Comment by escowles** [comment]
@awoods I agree that we should use the current editorial process for updating the ontology — I was thinking of the mechanics of publishing updates.  I believe the process that has been used for the Fedora 4 ontology will be fine, and it would be good to get the current ontology, such as it is, published in draft form as soon as possible.

**Comment by zimeon** [comment]
Discussion in API Call today resulted in decision to use http://fedora.info/definitions/fcrepo# namespace and `fcrepo` as prefered prefix.

**Comment by acoburn** [comment]
This issue is _not_ resolved. Could someone please reopen this issue? This issue will be resolved when the Fedora ontology is published at http://fedora.info/definitions/fcrepo#

**Comment by awoods** [comment]
resolved
